### ISSUE TITLE
\subsubsection highlight in ToC

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#257]
Originally reported by: **Danilo Galisteu (Bitbucket: [danilogalisteu](https://bitbucket.org/danilogalisteu), GitHub: [danilogalisteu](https://github.com/danilogalisteu))**

----------------------------------------

From the documentation (10.5 Adding a Table of Contents):
```
#!latex
subsectionstyle=⟨style for current subsection⟩/
⟨style for other subsections in current section⟩/
⟨style for subsections in other sections⟩

subsubsectionstyle=⟨style for current subsubsection⟩/
⟨style for other subsubsections in current subsection⟩/
⟨style for subsubsections in other subsections⟩
```
Comparing those two, I believe there should be another level of style for SubSubSections. The way it is right now, when I define:
```
#!latex
subsubsectionstyle=show/show/show
```
then subsubsections from other sections are also highlighted, which is certainly undesirable, and when I define:
```
#!latex
subsubsectionstyle=show/show/shade
```
then subsubsections from the same section are also shaded. The ideal definition would be:
```
#!latex
subsubsectionstyle=⟨style for current subsubsection⟩/
⟨style for other subsubsections in current subsection⟩/
⟨style for subsubsections in other subsections in current section⟩/
⟨style for subsubsections in other subsections in other sections⟩
```
with four levels of style, so that I can rightily define:
```
#!latex
subsubsectionstyle=show/show/show/shade
```
and get the highlight only inside the current section.

----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/257

**Comment by josephwright** [comment]
*Original comment by* **Joseph Wright (Bitbucket: [josephwright](https://bitbucket.org/josephwright), GitHub: [josephwright](https://github.com/josephwright))**:

----------------------------------------

This is one of those things that reflects the fact that Till really doesn't like `\subsubsection` in presentations, and while they do work in `beamer` the support is very limited. If you read over the code you'll find that the section/subsection stuff is hard coded and would be difficult to make more general. I'm minded to WONTFIX here (sort-of by design situation), but I do see the problem as `\subsubsection` does work. I'll take another look at this code before I make a decision.

**Comment by josephwright** [comment]
*Original comment by* **Michael Crouch (Bitbucket: [creidieki](https://bitbucket.org/creidieki), GitHub: [creidieki](https://github.com/creidieki))**:

----------------------------------------

subsubsectionstyle actually already does this; see the original patch at http://sourceforge.net/p/latex-beamer/patches/14/  .  It's working well for my presentation with show/show/show/shaded (I'm displaying a ToC outline at the beginning of each section, and I want that entire section to be highlighted).

The bug here is that the documentation doesn't list the 4th argument to subsubsectionstyle.

**Comment by josephwright** [comment]
*Original comment by* **Joseph Wright (Bitbucket: [josephwright](https://bitbucket.org/josephwright), GitHub: [josephwright](https://github.com/josephwright))**:

----------------------------------------

Document subsectionstyle behaviour correctly (fixes #257)

**Comment by josephwright** [comment]
*Original comment by* **Joseph Wright (Bitbucket: [josephwright](https://bitbucket.org/josephwright), GitHub: [josephwright](https://github.com/josephwright))**:

----------------------------------------

@creidieki Thanks for that: saves me trawling through the code!
